### PR TITLE
CG scaler on lightning, NF>0 for fire event

### PR DESCRIPTION
**Pull request by jkshuman**
Add scaler for cloud to ground lightning strikes, and strikes >0 for fire event

### Description:
Portion of strikes that are cloud to ground (CG) included in calculation of ignitions per Thonicke
et al 2010 and Latham and Williams (2001).
Calculation of ignitions moved to fire_intensity subroutine. Ignitions greater than zero included 
as condition for successful fire event. 
This address issue #589 

### Collaborators:
@rgknox @slevisconsulting @rosiealice @adrifoster 

### Expectation of Answer Changes:
This will change answers with fire active. Specifically, it will reduce fire area. 

### Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the in-code documentation .AND. (the [technical note](https://github.com/NGEET/fates-docs) .OR. the wiki) accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/master/CONTRIBUTING.md) document.
- [ ] FATES PASS/FAIL regression tests were run
- [x] If answers were expected to change, evaluation was performed and provided

### Test Results:
Simulations compiled and run across South America with and without fire. Comparison between prior default ignitions and scaled cloud to ground ignitions included.

Test Output:
Difference in Burned Fraction for year 4 (cloud to ground ignitions - all ignitions "default")
Ignitions_CG_ign-all_ign_yr4

### CODE REVIEW

**Comment by rgknox** [comment]
Glad to see that the calculation of burned area is contingent on having %fire = 1.  I think this was causing problems in the previous method (swapping routine order).

https://github.com/NGEET/fates/blob/20944a616455a032baeccea1bb0da9c00aec9bc5/fire/SFMainMod.F90#L752

File: fire/SFMainMod.F90 [hunk 1]
```diff
@@ -657,19 +658,30 @@ subroutine  fire_intensity ( currentSite )
...
     type(ed_site_type), intent(inout), target :: currentSite
-
     type(ed_patch_type), pointer :: currentPatch
 
     real(r8) ROS !m/s
     real(r8) W   !kgBiomass/m2
+    real(r8),parameter :: CG_strikes = .20_r8      !cloud to ground lightning strikes
+                                                   !Latham and Williams (2001)
+
+    currentSite%NF = 0.0_r8
```
Comment (rgknox) [inline fire/SFMainMod.F90]: This zero can be removed.

**Comment by jkshuman** [comment]
@slevisconsulting @pollybuotte link to these changes inside the read-lightning branch
https://github.com/jkshuman/fates/tree/FI-CG-read-lightning-tg130api8
I will confirm that this is also less in a bit.

**Comment by rgknox** [comment]
One minor suggestion about zero'ing. I think this PR looks good.  Note that adding %NF to the site structure will not require an addition to the restarts, since it is only used,  during fire calculations, and is calculated in that scheme as well.

**Comment by jkshuman** [comment]
@rgknox I am going to rework this again with the frac_burnt as part of calculating fire_intensity (FI) in a separate PR.  I will maintain this contingency in there as well. Yeah, that other implementation was a giant rabbit hole of problems...

File: main/EDInitMod.F90 [hunk 2]
```diff
@@ -280,6 +281,7 @@ subroutine set_site_properties( nsites, sites )
           sites(s)%dstatus = dstat
           
           sites(s)%acc_NI     = acc_NI
+          sites(s)%NF         = 0.0_r8
```
Comment (jkshuman) [inline main/EDInitMod.F90]: @rgknox just to confirm: you are saying to remove this line?

Comment (rgknox) [reply main/EDInitMod.F90]: I think this is fine, I don't think it will have an effect. 
Looks like you already fixed the line I was talking about.

Comment (jkshuman) [reply main/EDInitMod.F90]: OK. I was a little unclear on what was necessary in EDInit, so happy to take your input on if it needs to be changed.

File: fire/SFMainMod.F90 [hunk 3]
```diff
@@ -684,7 +695,7 @@ subroutine  fire_intensity ( currentSite )
           if( hlm_masterproc == itrue ) write(fates_log(),*) 'fire_intensity',currentPatch%fi,W,currentPatch%ROS_front
        endif
        !'decide_fire' subroutine shortened and put in here... 
-       if (currentPatch%FI >= fire_threshold) then  ! 50kW/m is the threshold for a self-sustaining fire
+       if (currentPatch%FI >= fire_threshold .and. currentSite%NF > 0._r8) then !50kW/m threshold for self-sustaining fire
```
Comment (slevisconsulting) [inline fire/SFMainMod.F90]: @jkshuman I see two new lines of code that sum up your changes in this pull request:
1) `currentSite%NF = ED_val_nignitions * years_per_day * CG_strikes` (currently line 680)
2) `if (currentPatch%FI >= fire_threshold .and. currentSite%NF > 0._r8) then` (currently line 698)
The rest is either infrastructure to support these changes or aesthetic improvements.

Assuming that your testing continues to pass, these changes seem fine to me.

**Comment by rgknox** [comment]
We can start testing this after review.  Would anyone else like to chime in and sign off?  @jkshuman, any other stake-holders you would like to check-in here?

**Comment by jkshuman** [comment]
Thanks for checking in @rgknox. I think with your review and @slevisconsulting we are good. This version is a basic update, so I am not worried about the logic.

**Comment by jkshuman** [comment]
these changes included in PR #594 . 
closing

**Pull request closed without merging**